src/main/java/dao/TicketDataBaseDaoImpl.java

```diff
@@ -2,125 +2,115 @@
 
 import model.BusTicket;
 import model.TicketDataBase;
-import util.ConnectionUtil;
-
-import java.sql.*;
-import java.util.ArrayList;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+import util.HibernateUtil;
 import java.util.List;
 
 public class TicketDataBaseDaoImpl implements TicketDataBaseDao {
-    private static final String SQL_INSERT = "INSERT INTO ticket_data_base (user_id, ticket_type, creation_date) VALUES (?, ?::tickettype, ?)";
-    private static final String SQL_SELECT = "SELECT * FROM ticket_data_base WHERE id = ?";
-    private static final String SQL_SELECT_TICKETS_USER_ID = "SELECT * FROM ticket_data_base INNER JOIN user_data_base ON ticket_data_base.user_id = user_data_base.id WHERE user_data_base.id = ?";
-    private static final String SQL_UPDATE_TICKET_TYPE = "UPDATE ticket_data_base SET ticket_type = ?::tickettype WHERE id = ?";
-    private static final String SQL_DELETE = "DELETE FROM ticket_data_base WHERE id = ?";
 
     @Override
     public TicketDataBase save(TicketDataBase ticketDataBase) {
-        try (Connection connection = ConnectionUtil.getConnection();
-             PreparedStatement statement = connection.prepareStatement(SQL_INSERT, Statement.RETURN_GENERATED_KEYS)) {
-            statement.setLong(1, ticketDataBase.getUserId());
-            statement.setString(2, ticketDataBase.getTicketType().name());
-            statement.setDate(3, java.sql.Date.valueOf(ticketDataBase.getCreationDate()));
-            int updatedRows = statement.executeUpdate();
-            if (updatedRows < 1) {
-                throw new RuntimeException("Inserted 0 rows");
+        SessionFactory sessionFactory = HibernateUtil.getSessionFactory();
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.save(ticketDataBase);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
             }
-            ResultSet generatedKeys = statement.getGeneratedKeys();
-            if (generatedKeys.next()) {
-                Long id = generatedKeys.getObject(1, Long.class);
-                ticketDataBase.setId(id);
+            throw new RuntimeException("Error while saving a ticket");
+        } finally {
+            if (session != null) {
+                session.close();
             }
```

Comment:
can we solve it using try with resources?

Reply:
Good morning !!
![Screenshot_9](https://github.com/user-attachments/assets/5ab4914b-f
![Screenshot_13](https://github.com/user-attachments/assets/9adf611b-a312-4f8f-8af0-3bc8b43ff327)
9fe-4e9e-b558-3e64bbcb7057)
 I can't use Telegram. Attackers hacked my account. I added  screnshots here.
And of course , I can use  try with resources)

Reply:
Can you add it?
Very sorry to hear that... but I thought so because I was written to and asked for money on your name .

and what is it? 
![image](https://github.com/user-attachments/assets/3706f8a8-f939-4f62-99c5-fe13ddc80059)

Reply:
Oh, sorry about this situation with money ((
These numbers are part of an unsuccessful screenshot, I apparently didn’t delete the whole thing (

Reply:
I need to use  try with resources in all transactions . Or change it only in lines which you showed ?

Reply:
in all please))

no problem))

Reply:
![image](https://github.com/user-attachments/assets/0079bcb6-6530-4647-b528-c9bf9322e681)
They didn't know we were communicating in English.
And as far as I understand you are not from Ukraine)))

Reply:
Yes , I live in Poland now)  I so angry about this situation. I'm so sorry! I hate such people !! I wrote to telegram support , and wait when they block my account.

Reply:
As I said, it's not a problem for me. It's not the first time) 
But I think some people could send them (
Where are you from in general? Or from Poland too?